augur/subsample.py

```diff
@@ -262,6 +311,83 @@ def _merge_options(sample_options: Dict[str, Any], defaults: Optional[Dict[str,
     return {**defaults, **sample_options}
 
 
+def _resolve_filepaths(options: Dict[str, Any], search_paths: List[Path], filepath_options: Set[str]) -> Dict[str, Any]:
+    """
+    Resolve filepath values in config options.
+    """
+    for key, value in options.items():
+        if key in filepath_options:
+            if isinstance(value, list):
+                options[key] = [_resolve_filepath_str(v, search_paths) for v in value]
+            elif isinstance(value, str):
+                options[key] = _resolve_filepath_str(value, search_paths)
+
+    return options
+
+
+def _resolve_filepath_str(path: str, search_paths: List[Path]) -> str:
+    """Resolve a string-based filepath by wrapping _resolve_filepath."""
+    return _resolve_filepath(Path(path), search_paths).as_posix()
```

Comment:
```suggestion
    return str(_resolve_filepath(Path(path), search_paths))
```

so that we don't assume POSIX paths? (Not that I think we support Augur on Windows, but…)

Reply:
:+1: One thing I meant to add but forget is: at the point of doing `str(fn(…))`, it seems not worth wrapping into a separate function at all.

Reply:
Removed the wrapper and switched to `str()` in various places:

https://github.com/nextstrain/augur/blob/25c4448a542290d905b4973d6acc2d7f4e3e6c20/augur/subsample.py#L381-L386

https://github.com/nextstrain/augur/blob/25c4448a542290d905b4973d6acc2d7f4e3e6c20/augur/subsample.py#L473-L474

https://github.com/nextstrain/augur/blob/25c4448a542290d905b4973d6acc2d7f4e3e6c20/augur/subsample.py#L484-L487